group_project_v2/mixins.py

```diff
@@ -125,7 +126,180 @@ def real_user_id(self, anonymous_student_id):
+            if xblock_settings and setting in xblock_settings:
+                result = xblock_settings[setting]
+        return result
+
+
+class AuthXBlockMixin(SettingsMixin, ProjectAPIXBlockMixin):
+
+    _DEFAULT_TA_ROLE = ('assistant',)
+
+    _SEE_DASHBOARD_ROLE_PERMS = "see_dashboard_role_perms"
+    _SEE_DASHBOARD_FOR_ALL_ORGS_ROLE_PERMS = "see_dashboard_for_all_orgs_perms"
+    _TA_ROLE = 'ta_roles'
+
+    @property
+    def see_dashboard_role_perms(self):
+        """
+        :return: Returns a list of group names, user needs to be a member of
+                 single group from this list to access dashboard.
+                 Additionally he will see this dashboard
+                 filtered so he only sees students from organisations he
+                 belongs to.
+        :rtype: Iterable[str]
+        """
+        return self._get_setting(self._SEE_DASHBOARD_ROLE_PERMS, [])
+
+    @property
+    def see_dashboard_for_all_orgs_perms(self):
+        """
+        :return: Returns a list of group names, user needs to be a member of
+                 single group from this list to access dashboard.
+                 Members of these roles will see dashboard with users
+                 from all organisations.
+        :rtype: Iterable[str]
+        """
+        return self._get_setting(self._SEE_DASHBOARD_FOR_ALL_ORGS_ROLE_PERMS, [])
+
+    @property
+    def ta_roles(self):
+        """
+
+        .. note:
+
+            This returns different thing than self.see_dashboard_for_all_orgs
+            this returns a **course** role, and rest contains a user group.
+
+        :return: Returns a list of group names. Normally to access a group work
+                 view user will need to belong to a team. If user is a member
+                 of this group he will be able to access group work for every
+                 group work team.
+        :rtype: Iterable[str]
+        """
+        return self._get_setting(self._TA_ROLE, self._DEFAULT_TA_ROLE)
+
+    def can_access_dashboard(self, user_id):
+        """
+        :param user_id:
+        :return: True if user can access dashboard.
+        """
+        user_groups = self._user_groups(user_id)
+        return bool(user_groups & self._access_dashboard_roles)
+
+    def check_dashboard_access(self, user_id):
+        """
+        :return: None
+        :raise GroupworkAccessDeniedError: If user can't access dashboard
+        """
+        if not self.can_access_dashboard(user_id):
+            raise GroupworkAccessDeniedError("User can't access dashboard")
+
+    def is_user_ta(self, user_id, course_id):
+        """
+        :return: True if user is a TA for a gven course
+        """
+
+        granted_roles = {r["role"] for r in self.project_api.get_user_roles_for_course(user_id, course_id)}
+        allowed_roles = set(self.ta_roles)
+        return bool(allowed_roles & granted_roles)
+
+    def check_outsider_allowed(self, user_id, course_id):
```

Comment:
My reasoning was that this method checks if outsider --- given as ``user_id`` -- is allowed, coincidentally this checks whether user is a TA, however this might not always be the case. 

Let me know if I convinced you :)